gnosis/gp-v2-services
# Extend Orders Table and Update existing Records

This PR addresses and closes #898 and closes #899 as follows:

1. introduce a tiny table `settlement_version` which holds `byeta` `settlement_contract` address
2. Introduce an enums BalanceTo and BalanceFrom (erc20 internal, external). We simutaneously introduce a rust enum DbBalanceTo/From
3. Alter orders table by adding three new columns (version_id, balance_from, balance_to) 
4. These columns are all filled in via default with appropriate values for all existing records. The columns are set to be NOT NULL.  and after existing records are updated, the default condition is dropped.
Note that we have also updated the raw sql insert order statement to start filling in these values as well. 

### Test Plan
No logical changes here, 

To manually check that the upgrade works, follow these steps:

1. Checkout main and run the orderbook (on a fresh or existing DB)
2. Run the Front end locally and place an order with the old schema
3. Shut down the orderbook service and checkout this branch
4. Apply the DB schema here and check that your order has been updated as expected.
